src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -577,35 +602,87 @@ sub-<label>/[ses-<label>/]
-file to obtain scanning parameters. \_epi files can be a 3D or 4D - in the
-latter case all timepoints share the same scanning parameters. To indicate which
-run is intended to be used with which functional or diffusion scan the
-IntendedFor field in the JSON file should be used.
+As for other EPI sequences, these field mapping sequences may have any of the
+[in-plane spatial encoding](#in-plane-spatial-encoding) metadata keys.
+However, please note that `PhaseEncodingDirection` and `TotalReadoutTime` keys
+are REQUIRED for these field mapping sequences.
+
+### Expressing the MR protocol intent for fieldmaps
+
+#### Using `B0FieldIdentifier` metadata
+
+The general purpose `B0FieldIdentifier` MRI metadata is RECOMMENDED for the
+prescription of the B0 field estimation intent of the original acquisition
+protocol.
+This encoding fully overrides the `IntendedFor` approach (see below).
+However, the `IntendedFor` encoding is permitted and RECOMMENDED in combination
+with the `B0FieldIdentifier` to maintain compatibility with outdated software.
+
+For example, a `sub-<label>_fieldmap.json` file denoting a fieldmap identifier
+`se_fmap0` could read:
+
+```JSON
+{
+  "Units": "rad/s",
+  "B0FieldIdentifier": "se_fmap0"
+}
+```
+
+#### Using `IntendedFor` metadata
+
+Fieldmap data can be linked to a specific scan(s) it was
+acquired for by filling the `IntendedFor` field in the corresponding JSON file.
+
+| **Key name** | **Requirement level** | **Data type**                         | **Description**                                                                                                                                                                                                                                                                |
+| ------------ | --------------------- | ------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| IntendedFor  | RECOMMENDED           | [string][] or [array][] of [string][] | Contains one or more filenames with paths relative to the participant subfolder. The path needs to use forward slashes instead of backward slashes. This field is OPTIONAL, and in case the fieldmaps do not correspond to any particular scans it does not have to be filled. |
+
+For example, extending from the previous example of `sub-<label>_fieldmap.json` file:
+
+```JSON
+{
+  "Units": "rad/s",
+  "B0FieldIdentifier": "se_fmap0",
+  "IntendedFor": "func/sub-01_task-motor_bold.nii.gz"
+}
+```
+
+or:
+
+```JSON
+{
+  "Units": "rad/s",
+  "B0FieldIdentifier": "se_fmap0",
+  "IntendedFor": [
+    "ses-pre/func/sub-01_ses-pre_task-motor_run-1_bold.nii.gz",
+    "ses-post/func/sub-01_ses-post_task-motor_run-1_bold.nii.gz"
```

Comment:
Just to see that I get this: In this case, there will be two separate files stored in the `fmap` folder, one with `sub-01_ses-pre` and one with `sub-01_ses-post` that are the field map files. What could the full filenames look like in this  case?

Reply:
hehehe, the PR needs to be much clearer...

So these two files are part of the `IntendedFor` field, which is what we already have in BIDS and says: the NIfTI to which this JSON file is associated is the fieldmap source for correcting these two other `_bold` files.

Reply:
Maybe the proposal is more readable here: https://bids-specification--622.org.readthedocs.build/en/622/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#expressing-the-mr-protocol-intent-for-fieldmaps